### PR TITLE
Add tests for functions without end marker. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Inspired by this downstream test in wabt: https://github.com/WebAssembly/wabt/pull/1775

Fixes: #1404

### CODE REVIEW

**Comment by sbc100** [comment]
I wonder if "section size mismatch" is wrong error here, but its the one that the interpreter currently generates.

**Review by rossberg — APPROVED** [review]
Well, the code section in this example is followed by a data section, which starts with the same opcode as END (I assume that's intentional?). So the decoder assumes the prior section size is wrong, not the contents. (The interpreter decodes each section until it is syntactically complete and then checks the current position against the declared size.)

**Comment by sbc100** [comment]
> I didn't pick data section to follow the code for any other reason other than its the only one that is valid after the code section (aside from custom sections). Odd coincidence the END and is the same byte (0xb) as the data sections code.
> 
> I will add a comment as to why error message from the interpreter its somewhat misleading.

I dropped the data section and used a second function instead to signal the early end of the first function.

**Comment by sbc100** [comment]
@rossberg would you consider enabling the "Auto-merge" feature on this repo?   I basically means I can have change like this merge if & when the tests all pass without having to remember to come back and check myself.

File: test/core/binary.wast [hunk 1]
```diff
@@ -368,6 +368,25 @@
   "integer too large"
 )
 
+;; Function with missing end marker
+(assert_malformed
+  (module binary
+    "\00asm" "\01\00\00\00"
+    "\03\02\01\00"             ;; Function section
```
Comment (rossberg) [inline test/core/binary.wast:375]: You may want to adjust this to have two versions as well.

Perhaps also add a type section, because the function type now actually points to an undefined type index, which may make most decoders bail out early, before even reaching the intended bug.

File: test/core/binary.wast [hunk 2]
```diff
@@ -368,6 +368,25 @@
...
+    "\0a\06\02"                ;; Code section with 2 functions
```
Comment (rossberg) [inline test/core/binary.wast:376]: The section length should be adjusted as well.

**Review by rossberg — COMMENTED** [review]
He he. It's actually an interesting corner case, so perhaps it's worth having both versions of the test? You could also add a third version where there is nothing past the missing end.

**Comment by rossberg** [comment]
As for auto-merge, I'm not sure it's a good idea for a repo like this. It's frequent practice that reviewers approve a PR despite small comments for change, in order to avoid another day-long turn-around across time zones. I have seen it happen too often that the auto-merge label is overlooked in such cases, causing a PR to be merged unintended.

**Comment by sbc100** [comment]
> As for auto-merge, I'm not sure it's a good idea for a repo like this. It's frequent practice that reviewers approve a PR despite small comments for change, in order to avoid another day-long turn-around across time zones. I have seen it happen too often that the auto-merge label is overlooked in such cases, causing a PR to be merged unintended.

True, I have noticed that too.  

It would be nice if there was way to for the auto-merge to only ever be triggered by the author and never by the approver.  I think that would avoid all the accidental merging (it would be the same as the power the author already has to land things with the current merge button but avoid them having to wait for tests).

i.e. "Auto-merge on test completion" as distinct from "Auto-merge on review approval" (that later being the dangerous on I think).

**Comment by sbc100** [comment]
Expanded to 3 separate tests:

- Missing end at EOF
- Missing end at function boundary
- Missing end at section boundary

**Review by rossberg — APPROVED** [review]
Excellent, thanks!
